### PR TITLE
Adprime: add userSync

### PR DESCRIPTION
**Pull request by Adprime**

### CODE REVIEW

**Comment by gargcreation1992** [comment]
@Adprime PR checks are failing. Please check

**Comment by SyntaxNode** [comment]
> @Adprime PR checks are failing. Please check

The pipeline failures look unrelated. I restarted the checks.

File: static/bidder-info/adprime.yaml [hunk 1]
```diff
@@ -12,3 +12,10 @@ capabilities:
       - banner
       - video 
       - native
+userSync:
+  redirect:
+    url: "https://sync.adprime.com/pbserver?gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&ccpa={{.USPrivacy}}&gpp={{.GPP}}&gpp_sid={{.GPPSID}}&redir={{.RedirectURL}}"
+    userMacro: "[UID]"
```
Comment (onkarvhanumante) [inline static/bidder-info/adprime.yaml:18]: redirect url working

```
https://sync.adprime.com/pbserver?gdpr=&gdpr_consent=&ccpa=&gpp=&gpp_sid=&redir=https%3A%2F%2Fprebid.adnxs.com%2Fpbs%2Fv1%2Fsetuid%3Fbidder%3Dadprime%26gdpr%3D%26gdpr_consent%3D%26f%3Di%26uid%3D%22%5BUID%5D%22
```

File: static/bidder-info/adprime.yaml [hunk 2]
```diff
@@ -12,3 +12,10 @@ capabilities:
...
+  iframe:
+    url: "https://sync.adprime.com/pbserverIframe?gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&ccpa={{.USPrivacy}}&gpp={{.GPP}}&gpp_sid={{.GPPSID}}&pbserverUrl={{.RedirectURL}}"
+    userMacro: "[UID]"
```
Comment (onkarvhanumante) [inline static/bidder-info/adprime.yaml:21]: @Adprime  iframe didn't append uid at the end for following url:

```
https://sync.adprime.com/pbserverIframe?gdpr=&gdpr_consent=&ccpa=&gpp=&gpp_sid=&pbserverUrl=https%3A%2F%2Fprebid.adnxs.com%2Fpbs%2Fv1%2Fsetuid%3Fbidder%3Dadprime%26gdpr%3D%26gdpr_consent%3D%26f%3Di%26uid%3D%22%5BUID%5D%22
```

image

looks like uuid is present in js script but not correctly appended to redirect url

Comment (Adprime) [reply static/bidder-info/adprime.yaml:21]: @onkarvhanumante 
is your link valid?
```
https://prebid.adnxs.com/pbs/v1/setuid?bidder=adprime&gdpr=&gdpr_consent=&f=i&uid="[UID]"
```

I corrected it a little
```
https://prebid.adnxs.com/pbs/v1/setuid?bidder=adprime&gdpr=&gdpr_consent=&f=i&uid=[UID]
```

in this case, when calling
```
https://sync.adprime.com/pbserverIframe?gdpr=&gdpr_consent=&ccpa=&gpp=&gpp_sid=&pbserverUrl=https%3A%2F%2Fprebid.adnxs.com%2Fpbs%2Fv1%2Fsetuid%3Fbidder%3Dadprime%26gdpr%3D%26gdpr_consent%3D%26f%3Di%26uid%3D%5BUID%5D
```

everything works correctly
image

Comment (onkarvhanumante) [reply static/bidder-info/adprime.yaml:21]: > @onkarvhanumante is your link valid?
> 
> ```
> https://prebid.adnxs.com/pbs/v1/setuid?bidder=adprime&gdpr=&gdpr_consent=&f=i&uid="[UID]"
> ```
> 
> I corrected it a little
> 
> ```
> https://prebid.adnxs.com/pbs/v1/setuid?bidder=adprime&gdpr=&gdpr_consent=&f=i&uid=[UID]
> ```
> 
> in this case, when calling
> 
> ```
> https://sync.adprime.com/pbserverIframe?gdpr=&gdpr_consent=&ccpa=&gpp=&gpp_sid=&pbserverUrl=https%3A%2F%2Fprebid.adnxs.com%2Fpbs%2Fv1%2Fsetuid%3Fbidder%3Dadprime%26gdpr%3D%26gdpr_consent%3D%26f%3Di%26uid%3D%5BUID%5D
> ```
> 
> everything works correctly image



@Adprime  `"[UID]"` was changed to `[UID]` in your case. But `static/bidder-info/adprime.yaml` has `"[UID]"` as userMacro

Comment (Adprime) [reply static/bidder-info/adprime.yaml:21]: @onkarvhanumante as far as I understand, we must pass a **string** to `userMacro`. The .yaml specification allows you to specify a string with or without quotes (`"`). If you remove the `"`, the `UID` string will be enclosed in `[` and `]`, which is a **list**. Please correct me if I'm wrong

Comment (Adprime) [reply static/bidder-info/adprime.yaml:21]: @onkarvhanumante any updates?

**Review by onkarvhanumante — APPROVED** [review]

**Review by gargcreation1992 — APPROVED** [review]

**Pull request merged**